internal/indexer/symbols.go

```diff
@@ -0,0 +1,184 @@
+package indexer
+
+import (
+	"fmt"
+	"go/ast"
+	"go/doc"
```

Comment:
There seems to be some historical precedent for why we should not use go/doc for generating document symbols from the go-langserver days: https://github.com/sourcegraph/go-langserver/commit/00cb77287a8d0f185e5cb978a7c75197609b9f54#diff-eb4c6c7112f00549c81d7f1a012322112eaf0d38f3d46035e89a8b323779923b

I can also imagine the go/doc interface becoming a limiting factor in the future i we try to support bi-directional linking of symbols, such as when Go generics arrive. I will look into whether or not it makes sense to reuse what we've done historically in go-langserver instead.